### PR TITLE
T1075 pass the hash

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature
> Pass the hash
> Helper file in UI for common technique report functions

image

### CODE REVIEW

File: monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+export function RenderMachine(val){
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/ui/src/components/attack/techniques/Helpers.js]: start function name with lower case letter

File: monkey/monkey_island/cc/ui/src/components/attack/techniques/T1075.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import ReactTable from "react-table";
+import { RenderMachine } from "./Helpers"
+
+
+class T1075 extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.props.data.successful_logins.forEach((login) => {
+      if(login.attempts[0].ntlm_hash !== ""){
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/ui/src/components/attack/techniques/T1075.js]: export to function `setLoginHashType`

File: monkey/monkey_island/cc/services/attack/technique_reports/T1075.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    tech_id = "T1075"
+    unscanned_msg = "Monkey didn't try to use pass the hash attack."
+    scanned_msg = "Monkey tried to use hashes while logging in but didn't succeed."
+    used_msg = "Monkey successfully used hashed credentials."
+
+    login_attempt_query = {'data.attempts': {'$elemMatch': {'$or': [{'ntlm_hash': {'$ne': ''}},
+                                                                    {'lm_hash': {'$ne': ''}}]}}}
+
+    # Gets data about successful PTH logins
+    query = [{'$match': {'telem_type': 'exploit',
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1075.py]: telem_catagory

File: monkey/monkey_island/cc/services/attack/technique_reports/T1075.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                           'attempts': {'$filter': {'input': '$data.attempts',
+                                                    'as': 'attempt',
+                                                    'cond': {'$eq': ['$$attempt.result', True]}}}}}]
+
+    @staticmethod
+    def get_report_data():
+        data = {'title': T1075.technique_title(T1075.tech_id)}
+        successful_logins = list(mongo.db.telemetry.aggregate(T1075.query))
+        data.update({'successful_logins': successful_logins})
+        if successful_logins:
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1075.py]: now we have a function getting message and status

File: monkey/monkey_island/cc/ui/src/components/attack/techniques/T1075.js (hunk 2 above)
Comment (VakarisZ) [reply monkey/monkey_island/cc/ui/src/components/attack/techniques/T1075.js]: Check if I've done it correctly

File: monkey/monkey_island/cc/services/attack/technique_reports/T1075.py [hunk 5]
```diff
@@ -35,9 +35,9 @@ def get_report_data():
         successful_logins = list(mongo.db.telemetry.aggregate(T1075.query))
         data.update({'successful_logins': successful_logins})
         if successful_logins:
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1075.py]: set status according to successful_logins and count_documents(query),
afterwards use only 1 call to get_message_and_status

File: monkey/monkey_island/cc/services/attack/technique_reports/T1110.py [hunk 6]
```diff
@@ -35,11 +35,11 @@ def get_report_data():
                 result['successful_creds'].append(T1110.parse_creds(attempt))
 
         if succeeded:
-            data = T1110.get_message_and_status(T1110, ScanStatus.USED)
+            data = T1110.get_message_and_status(ScanStatus.USED)
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1110.py]: Like on T1075, set status according to succeeded and attempts, call get_message_and_status only once

File: monkey/monkey_island/cc/services/attack/technique_reports/__init__.py [hunk 7]
```diff
@@ -47,51 +47,64 @@ def get_report_data():
         pass
 
     @staticmethod
-    def technique_status(technique):
+    def technique_status(tech_id):
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/__init__.py]: make `@classmethod`
get actual technique and not tech_id
inside function get tech_id from technique

File: monkey/monkey_island/cc/services/attack/technique_reports/__init__.py [hunk 8]
```diff
@@ -47,51 +47,64 @@ def get_report_data():
...
         elif status == ScanStatus.SCANNED:
-            return technique.scanned_msg
+            return cls.scanned_msg
         else:
-            return technique.used_msg
+            return cls.used_msg
 
     @staticmethod
-    def technique_title(technique):
+    def technique_title(tech_id):
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/__init__.py]: Same comment as in technique_status

File: monkey/monkey_island/cc/services/attack/technique_reports/T1210.py [hunk 9]
```diff
@@ -14,15 +14,16 @@ class T1210(AttackTechnique):
...
         exploited_services = T1210.get_exploited_services()
         if exploited_services:
-            data.update({'status': ScanStatus.USED.name, 'message': T1210.used_msg})
+            status = ScanStatus.USED
         elif scanned_services:
-            data.update({'status': ScanStatus.SCANNED.name, 'message': T1210.scanned_msg})
+            status = ScanStatus.SCANNED
         else:
-            data.update({'status': ScanStatus.UNSCANNED.name, 'message': T1210.unscanned_msg})
+            status = ScanStatus.UNSCANNED.name
```
Comment (itaymmguardicore) [inline monkey/monkey_island/cc/services/attack/technique_reports/T1210.py]: remove '.name'

Comment (VakarisZ) [reply monkey/monkey_island/cc/services/attack/technique_reports/T1210.py]: Nice catch, thanks

**Pull request merged**